graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/relay/PaginationHelper.java

```diff
@@ -154,10 +161,28 @@ private static <T> List<T> collectItems(Iterator<T> it, Arguments arguments, Cur
             if (arguments.before != null && cursorSupport.getCursor(value).equals(arguments.before)) {
                 break;
             }
+
+            //Adding a limit of 1000 to avoid OOM
+            if(count.intValue() == 500) {
+                logger.warn("The current paginated query is returning more than 500 items. This may cause a memory leak.");
+            } else if(count.intValue() == nodeLimit) {
+                logger.warn("The current paginated query is returning more than {} items. Stopping the query.", nodeLimit);
+                break;
+            }
         }
         return new ArrayList<>(items);
     }
 
+    private static int getNodeLimit() {
+        BundleContext bundleContext = FrameworkUtil.getBundle(DXGraphQLConfig.class).getBundleContext();
+        ServiceReference<DXGraphQLConfig> dxGraphQLConfigServiceReference = bundleContext.getServiceReference(DXGraphQLConfig.class);
+        DXGraphQLConfig service = bundleContext.getService(dxGraphQLConfigServiceReference);
+        if(service!= null) {
+            return service.getNodeLimit();
+        }
+        return 5000;
+    }
+
```

Comment:
We could store the config value in a static thread safe integer like AtomicInteger:

```
private static AtomicInteger nodeLimit = new AtomicInteger(5000);

    public static void updateLimit(int limit) {
        nodeLimit.set(limit);
    }

    public static int getNodeLimit() {
        return nodeLimit.get();
    }
```

To avoid search for OSGI service of the conf on every threads, the conf could reload the static limit on update safely, what do you think ?